src/main/java/com/google/devtools/build/lib/rules/java/JavaCommon.java

```diff
@@ -374,6 +374,8 @@ private ImmutableList<String> computeJavacOpts(Collection<String> extraRuleJavac
     return ImmutableList.<String>builder()
         .addAll(javaToolchain.getJavacOptions(ruleContext))
         .addAll(extraRuleJavacOpts)
+        // Consumed by com.google.devtools.build.runfiles.AutoBazelRepositoryProcessor.
+        .add("-Abazel.repository=" + ruleContext.getRepository().getName())
```

Comment:
@cushon asks:

>>> Instead of unconditionally passing it to all builds, could it be passed to builds that use AutoBazelRepository?
>>
>> (me) how would one check whether a build is using AutoBazelRepository?
>
> It's just a java_plugin, right? So they'd have to depend on it, and com.google.devtools.build.runfiles.AutoBazelRepositoryProcessor would be in the list of processors for the compilation, and so on.

Reply:
I have something that compiles and will run the tests, but I don't know whether I'm holding it right. (Seems like `RunfilesLibraryInfo` made a return in the form of an annotation processor :-)).

Reply:
@cushon Now that I'm digging into this deeper, how is it decided whether a plugin is "API generating" or not? I am not sure whether I would need to pass in `-Abazel.repository` for header compilation actions.

Reply:
Added a commit that implements this and has the test passing.

Reply:
> it does generate API (a new class), even if we don't really want dependents to use the generated class

I think we probably want `generates_api = False`, unless we actually want dependents to be able to use the generated class, or decide it's not worth trying to discourage.

Using the generated code in the same target is fine, `generates_api = True` is for cases that deliberately support other targets directly accessing the generated code.

Reply:
Okay, then `generates_api = False` sounds right. We really want the generated class to be used only from the annotated class.